specs/altair/light-client/full-node.md

```diff
@@ -103,42 +111,30 @@ def create_light_client_update(state: BeaconState,
     assert hash_tree_root(attested_header) == hash_tree_root(attested_block.message) == block.message.parent_root
     update_attested_period = compute_sync_committee_period_at_slot(attested_block.message.slot)
 
+    update = LightClientUpdate()
+
+    update.attested_header = block_to_light_client_header(attested_block)
+
     # `next_sync_committee` is only useful if the message is signed by the current sync committee
     if update_attested_period == update_signature_period:
-        next_sync_committee = attested_state.next_sync_committee
-        next_sync_committee_branch = compute_merkle_proof_for_state(attested_state, NEXT_SYNC_COMMITTEE_INDEX)
-    else:
-        next_sync_committee = SyncCommittee()
-        next_sync_committee_branch = [Bytes32() for _ in range(floorlog2(NEXT_SYNC_COMMITTEE_INDEX))]
+        update.next_sync_committee = attested_state.next_sync_committee
+        update.next_sync_committee_branch = \
+            compute_merkle_proof_for_state(attested_state, NEXT_SYNC_COMMITTEE_INDEX)
 
     # Indicate finality whenever possible
     if finalized_block is not None:
         if finalized_block.message.slot != GENESIS_SLOT:
-            finalized_header = BeaconBlockHeader(
-                slot=finalized_block.message.slot,
-                proposer_index=finalized_block.message.proposer_index,
-                parent_root=finalized_block.message.parent_root,
-                state_root=finalized_block.message.state_root,
-                body_root=hash_tree_root(finalized_block.message.body),
-            )
-            assert hash_tree_root(finalized_header) == attested_state.finalized_checkpoint.root
+            update.finalized_header = block_to_light_client_header(finalized_block)
+            assert hash_tree_root(update.finalized_header) == attested_state.finalized_checkpoint.root
         else:
             assert attested_state.finalized_checkpoint.root == Bytes32()
-            finalized_header = BeaconBlockHeader()
-        finality_branch = compute_merkle_proof_for_state(attested_state, FINALIZED_ROOT_INDEX)
-    else:
-        finalized_header = BeaconBlockHeader()
-        finality_branch = [Bytes32() for _ in range(floorlog2(FINALIZED_ROOT_INDEX))]
```

Comment:
Note to reviewers: removed because they are the default value of `update.finalized_header` and `update.finality_branch`.